nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1755,6 +1772,79 @@ mod test {
         logctx.cleanup_successful();
     }
 
+    #[test]
+    fn test_dataset_settings_modified_in_place() {
+        static TEST_NAME: &str = "planner_dataset_settings_modified_in_place";
```

Comment:
I added this test to give an example of "what the diff looks like when a dataset is modified".

`planner_dataset_settings_modified_in_place_1_2.txt` contains this example